zbus_macros/src/lib.rs

```diff
@@ -289,17 +289,17 @@ pub fn proxy(attr: TokenStream, item: TokenStream) -> TokenStream {
 /// using this since it will force all interested peers to fetch the new value and hence result in
 /// excess traffic on the bus.
 ///
-/// The method and property getter arguments support the following `zbus` attributes:
+/// The method and property arguments support the following `zbus` attributes:
 ///
 /// * `object_server` - This marks the method argument to receive a reference to the
 ///   [`ObjectServer`] this method was called by.
 /// * `connection` - This marks the method argument to receive a reference to the [`Connection`] on
 ///   which the method call was received.
 /// * `header` - This marks the method argument to receive the message header associated with the
-///   D-Bus method call being handled. For property getter methods, this will be an
-///   `Option<Header<'_>>`, which will be `None` when the function is being called as part of the
-///   initial object setup (before it gets registered on the bus), or when we send out property
-///   changed notifications.
+///   D-Bus method call being handled. For property methods, this will be an `Option<Header<'_>>`,
+///   which will be `None` for getters when the function is being called as part of the initial
```

Comment:
So currently it is always `Some` for setters? If so, here it's not clear why they're getting an `Option` in that case. I guess we want it to be an `Option` for #898.

I'd suggest removing all the details here and just say that: "which will be set to `None` if the method is called for reasons other than to respond to an external property access".

Reply:
Yes, it's always `Some` for setters.

TBH, I've done it that way at first because I wasn't sure I wasn't going to hit a case where `None` would be required, and ended up just keeping it for symmetry, so as much as I'd like to take credit for foresight related to #898, I can't. :smile:

But it's good to know it's useful that it stayed and `Option`.

Right, I will update the doc comments. Thanks!